corehq/apps/locations/static/locations/js/location_tree.async.js

```diff
@@ -146,4 +164,103 @@ function LocationModel(data, root, depth) {
         hqImport('analytix/js/google').track.event('Organization Structure', '+ New _______');
         return true;
     };
+
+    archive_loc = function(button, name, loc_id) {
```

Comment:
It may not look like it, but the way these functions are defined here, they're still actually being defined in the global namespace. I think if you define them as `this.archive_loc`, because of the way the bindings are set up, they'll still work (but be taking them from the `$data`/context object rather than from the global namespace.